external/fv3fit/fv3fit/emulation/compositions/blended.py

```diff
@@ -0,0 +1,54 @@
+from typing import Callable, Mapping
+import tensorflow as tf
+
+from fv3fit.keras.adapters import ensure_dict_output
+
+TensorDict = Mapping[str, tf.Tensor]
+
+
+def blended_model(
+    model_a: tf.keras.Model,
+    model_b: tf.keras.Model,
+    mask_func: Callable[[TensorDict], tf.Tensor],
+    inputs: Mapping[str, tf.keras.Input],
+) -> tf.keras.Model:
+    """
+    Take two models and blend their outputs given a mask function.
+
+    The masking operation is broadcastable so take care to ensure
+    that large tensors aren't created by omitting a feature
+    dimension from the mask, when singleton feature fields exist.
+
+    Args:
+        model_a: model to use at points where mask == True
+        model_b: model to use at points where mask == False
+        mask_func: masking function determining model output selection.  the
+            function uses the model inputs and should return a mask tensor
+            with [sample, feature] or [sample, 1] dimensions
+        inputs: input tensors for the model, assumed to be
+            cover all necessary inputs for model_a and model_b
+    """
+
+    # avoid name conflicts in blended model
+    model_a._name = "ModelA"
```

Comment:
Ditto about labeling with "mask_true/false" over "A/B"- it might make debugging easier.